fetch/schema_creation.go

```diff
@@ -0,0 +1,258 @@
+    LEFT JOIN pg_catalog.pg_index ix ON c.oid = ix.indrelid AND a.attnum = ANY(ix.indkey)
+    WHERE
+        c.relkind = 'r'  -- 'r' indicates a table (relation)
+        AND a.attnum > 0 -- Exclude system columns
+        AND c.relname = $1
+        AND c.relnamespace::regnamespace::text = $2
+) t1
+LEFT JOIN (
+    SELECT
+        c.column_name,
+        c.table_name,
+        c.table_schema,
+        c.udt_name,
+        t.definition AS udt_def,
+        c.ordinal_position
+    FROM
+        information_schema.columns c
+    LEFT JOIN (
+        SELECT
+            'CREATE TYPE IF NOT EXISTS ' || t.typname || ' AS ENUM ' ||
+            '(' || string_agg(quote_literal(e.enumlabel), ', ' ORDER BY e.enumsortorder) || ');' AS definition,
+            t.typname
+        FROM
+            pg_type t
+        JOIN pg_enum e ON t.oid = e.enumtypid
+        GROUP BY
+            t.typname
+    ) t ON c.udt_name = t.typname
+    WHERE
+        c.table_name = $1 AND c.table_schema = $2
+) t2 ON t1.column_name = t2.column_name
+    AND t1.table_name = t2.table_name
+    AND t1.schema_name = t2.table_schema
+ORDER BY
+    t1.table_name,
+    t2.ordinal_position;
+`
+	)
+
+	res := make([]columnWithType, 0)
+	logger.Info().Msgf("getting column types for table: %s", table.String())
+
+	switch conn := conn.(type) {
+	case *dbconn.PGConn:
+		rows, err := conn.Query(ctx, pgQuery, table.Table, table.Schema)
+		if err != nil {
+			return nil, err
+		}
+		for rows.Next() {
+			newCol := columnWithType{}
+			if err := rows.Scan(&newCol.schemaName, &newCol.tableName, &newCol.columnName, &newCol.dataType, &newCol.typeOid, &newCol.notNullable, &newCol.isPrimaryKey, &newCol.udtName, &newCol.udtDefinition); err != nil {
+				return nil, errors.Wrap(err, "failed to scan query result to a columnWithType object")
+			}
+			logger.Debug().Msgf("collect column:%s", newCol.String())
+			res = append(res, newCol)
+		}
+		logger.Info().Msgf("finished getting column types for table: %s", table.String())
+	// TODO(janexing): support mysql.
+	default:
+		return nil, errors.New("not supported conn type")
+	}
+
+	return res, nil
+}
+
+func GetDropTableStmt(table dbtable.DBTable) (string, error) {
+	tName, err := parser.ParseQualifiedTableName(table.Table.String())
+	if err != nil {
+		return "", err
+	}
+	res := tree.DropTable{
+		Names:    tree.TableNames{*tName},
+		IfExists: true,
+	}
+
+	return res.String(), nil
+}
+
+func GetCreateTableStmt(
+	ctx context.Context, logger zerolog.Logger, conn dbconn.Conn, table dbtable.DBTable,
+) (string, error) {
+	newCols, err := GetColumnTypes(ctx, logger, conn, table)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed get columns for target table: %s", table.String())
+	}
+
+	var res string
+	for _, col := range newCols {
+		if col.udtDefinition != "" {
+			logger.Warn().Msgf("the original schema contains enum type %q, a tentative enum type creation is triggered via %q", col.udtName, col.udtDefinition)
```

Comment:
Hmm. Im thinking this one would be wroth an info log.
```suggestion
			logger.Warn().Msgf("the original schema contains enum type %q. A tentative enum type will be created as %q", col.udtName, col.udtDefinition)
```